pkg/network/aci_network.go

```diff
@@ -128,23 +128,27 @@ func (pn *ProviderNetwork) setupNetwork(ctx context.Context, azConfig *auth.Conf
 	var rawResponse *http.Response
 	ctxWithResp := runtime.WithCaptureResponse(ctx, &rawResponse)
 
-	createSubnet := true
+	updateSubnet := false
```

Comment:
A better version would be
var updateSubnet, createNewSubnet bool

They will be initialized to 0. which is false.